src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -338,6 +312,163 @@ private AddTxResult AddCore(Transaction tx, bool isPersistentBroadcast, bool isR
+                        {
+                            // transaction which was on last position in sorted TxPool and was deleted to give
+                            // a place for a newly added tx (with higher priority) is now removed from hashCache
+                            // to give it opportunity to come back to TxPool in the future, when fees drops
+                            _hashCache.Delete(removed.Hash);
+                        }
+                    }
+                    isKnown |= !inserted;
+                }
+            }
+
+            if (!isKnown)
+            {
+                isKnown |= !isReorg && (_txStorage.Get(tx.Hash) is not null);
+            }
+
+            if (isKnown)
+            {
+                // If transaction is a bit older and already known then it may be stored in the persistent storage.
+                Metrics.PendingTransactionsKnown++;
+                if (_logger.IsTrace) _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, already known.");
+                return AddTxResult.AlreadyKnown;
+            }
+            
+            HandleOwnTransaction(tx, isPersistentBroadcast);
+            NotifySelectedPeers(tx);
+            StoreTx(tx);
+            NewPending?.Invoke(this, new TxEventArgs(tx));
+            return AddTxResult.Added;
+        }
+
+        private void UpdateBucket(Address senderAddress)
+        {
+            var bucketSnapshot = _transactions.GetBucketSnapshot(senderAddress);
+            
+            if (bucketSnapshot.Length == 0)
+            {
+                return;
+            }
+            
+            Account? account = _stateProvider.GetAccount(senderAddress);
+            UInt256 balance = account?.Balance ?? UInt256.Zero;
+            long currentNonce = (long)(account?.Nonce ?? UInt256.Zero);
+
+            UpdateGasBottleneck(bucketSnapshot, currentNonce, balance);
+        }
+
+        private void UpdateGasBottleneck(IReadOnlyList<Transaction> bucketSnapshot, long currentNonce, UInt256 balance)
+        {
+            Transaction tx = bucketSnapshot[0];
+            UInt256 previousTxBottleneck = tx.CalculatePayableGasPrice(_specProvider.GetSpec().IsEip1559Enabled, CurrentBaseFee, balance);
+
+            for (int i = 0; i < bucketSnapshot.Count; i++)
+            {
+                tx = bucketSnapshot[i];
+                UInt256 gasBottleneck = 0;
+
+                if (tx.Nonce == currentNonce + i)
+                {
+                    UInt256 effectiveGasPrice = tx.CalculateEffectiveGasPrice(_specProvider.GetSpec().IsEip1559Enabled, CurrentBaseFee);
+                    gasBottleneck = UInt256.Min(effectiveGasPrice, previousTxBottleneck);
+                }
+                
+                _transactions.NotifyChange(tx.Hash, tx, t =>
+                {
+                    tx.GasBottleneck = gasBottleneck;
+                });
+                
+                previousTxBottleneck = gasBottleneck;
+            }
+        }
```

Comment:
Maybe do a NotifyChange on whole bucket? Similarly to: https://github.com/NethermindEth/nethermind/blob/MEV_02/src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs#L222-L242